docs/developer-docs/daos/nns/proposal-advanced.mdx

```diff
@@ -0,0 +1,234 @@
+
+The NNS assigns a unique identity to each proposal that it receives.
+
+## Proposal topics & types
+
+Each proposal is of a _proposal type_, which determines what happens if the proposal is adopted or rejected. Technically this defines which canister method is invoked with which arguments.
+Each type of proposal belongs to a specific _proposal topic_. Topics are used for [following](TODO) and also determine some other details about how the proposal is processed.
+For example, the voting period and the voting reward weight are defined per topic.
+
+We present the proposals grouped by their topics, with the corresponding proposal types that they contain.
+
+### Topic NetworkCanisterManagement
+Installing and upgrading _system canisters_ that belong to the network.
+This includes the NNS DAO canisters, such as the governance and root canister, the ICP ledger canister with archive and index,
+other canister on the system subnets such as Internet Identity (II),
+the ckBTC, ckETH, and ckERC20 canisters,
+and the Wasm version of the SNS canisers (governance, root, ledger, swap). // TODO: link to SNS
+
+* NnsCanisterUpgrade: A proposal to upgrade an existing system canister. This proposal type is executed by the root canister. Beyond upgrading the Wasm module of the target canister, the proposal can also set the authorization information and the allocations.
+* NnsCanisterInstall: A proposal to add a new canister to be installed and executed in the NNS subnetwork. The root canister, which controls all canisters on the NNS except for itself, handles this proposal type. The call also expects the Wasm module that shall be installed.
+* NnsRootUpgrade: A proposal to upgrade the root canister in the NNS subnetwork. The proposal is processed by the Lifeline canister, which controls the root canister. The proposal updates the Wasm module.
+* HardResetNnsRootToVersion: Uninstall and Install Root with the WASM provided in the function.  If InitArgs are provided They will be passed to the canister_init function of the WASM provided. This function is meant as a Break Glass mechanism for when an open call context in the Root canister is preventing root or another canister from upgrading.
+* StopOrStartNnsCanister: Stop or start an NNS canister. Usually this is done as part of a canister upgrade, but this proposal offers the flexibility to just start or stop a canister if needed.
+
+The following are specific for SNS canisters.
+* AddSnsWasm: Add a new SNS canister WASM to SNS-W. All SNS DAOs can then upgrade to new version along the upgrade path defined in SNS-W. //TODO: (link to SNS page).
+* InsertSnsWasmUpgradePathEntries: Insert custom upgrade path entries into SNS-W for all SNSes, or for an SNS specified by its governance canister ID.
+The following proposal is specific for the bitcoin canister:
+* BitcoinSetConfig: //TODO: https://dashboard.internetcomputer.org/proposal/129394
+
+**How to verify NNS core canister upgrades**
```

Comment:
The idea would be that each known neuron has some topics that they specialise on.
As I assume that they would be one of the main consumers of this documentation, I thought it would be nice to keep all the information regarding one topic together. What do you think?

Reply:
That makes sense then, let's leave it as is.